provider/pkg/provider/provider.go

```diff
@@ -605,9 +605,27 @@ func (k *kubeProvider) Configure(_ context.Context, req *pulumirpc.ConfigureRequ
 		k.defaultNamespace = defaultNamespace
 	}
 
+	var kubeClientSettings KubeClientSettings
+	if obj, ok := vars["kubernetes:config:kubeClientSettings"]; ok {
+		err := json.Unmarshal([]byte(obj), &kubeClientSettings)
```

Comment:
Hmm I didn't realize nesting the config makes it so config would have to be through json. I wonder if it beats the purpose then - since it gets quite painful for users to configure. I guess adding environment variable defaults becomes increasingly important but I am not thrilled that this makes the `pulumi config set` workflow more painful.
I wonder if we should consider reverting to flattening this?

Reply:
The JSON is from the RPC interface. Configuration through the CLI requires the `--path` argument, but isn't too difficult:

```
pulumi config set --path kubernetes:kubeClientSettings.burst 30
```

Which results in the following stack config:

```yaml
config:
  kubernetes:kubeClientSettings:
    burst: 30
```

Note that this also only applies to the default provider. It's natural to specify in code for a first-class Provider:

```ts
const provider = new k8s.Provider("test", {
    kubeClientSettings: {
        burst: 20,
        qps: 20.5,
    }
});
```

Reply:
Ah gotcha. TIL!